src/app/components/demo-table/demo-table.component.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable no-undef */
+import { Component, OnInit } from '@angular/core';
+import { SwaggerService } from 'src/app/services/swagger.service';
+import { SwaggerSummary } from '../../models/swagger-summary/swagger-summary';
+import { ResultSummary } from '../../models/result-summary/result-summary';
+
+@Component({
+  selector: 'app-demo-table',
+  templateUrl: './demo-table.component.html',
+  styleUrls: ['./demo-table.component.scss']
+})
+export class DemoTableComponent implements OnInit {
+  constructor(private swaggerService: SwaggerService) { }
+
+  public swaggerSummary : SwaggerSummary;
+
+  public swaggerSummaryId : number;
+
+  public resultSummary : ResultSummary[];
+
+  async ngOnInit(): Promise<SwaggerSummary> {
+    this.swaggerSummaryId = Number(sessionStorage.getItem('swaggerSummaryId'));
+    const re = await this.swaggerService.loadSummaryInTable(this.swaggerSummaryId);
+    this.swaggerSummary = re;
+    this.resultSummary = re.resultsummaries;
+    // sessionStorage.removeItem('swaggerSummaryId');
+    return this.swaggerSummary;
```

Comment:
Why does this return anything? I don't believe `ngOnInit` is ever called directly, which means the return value is never used. At best, this should be `Promise<void>`.

While `async/await` with `ngOnInit` works just fine, it has several other implications in terms of the view being fully loaded before the data within the component has been processed. This can have some interference with `ngIf` and others, if not used carefully.

With this in mind, it is often recommended to keep `ngOnInit` as a regular method, which would ultimately, be just `void` return type.